tempest2/src/main/kotlin/app/cash/tempest2/Scan.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.tempest2
+
+import software.amazon.awssdk.enhanced.dynamodb.Expression
+
+interface Scannable<K : Any, I : Any> {
+  /**
+   * Scans up to the [pageSize] items or a maximum of 1 MB of data. This limit applies before the
+   * filter expression is evaluated.
+   */
+  fun scan(
```

Comment:
used to support scanning by segments. this doesn't exist in the 2.x SDK

Reply:
HUGE REGRESSION! OMG!

Reply:
(nothing to do here though)